app/src/main/java/org/opencord/aaa/impl/StateMachine.java

```diff
@@ -558,14 +622,57 @@ public void logoff() {
         private final Logger log = getLogger(getClass());
         private String name = "UNAUTHORIZED_STATE";
 
+        @Override
         public void start() {
             log.info("Moving from UNAUTHORIZED state to STARTED state.");
         }
 
+        @Override
         public void logoff() {
             log.info("Moving from UNAUTHORIZED state to IDLE state.");
         }
     }
 
 
+    /**
+     * Class for cleaning the StateMachine for those session for which no response is coming--implementing timeout.
+     */
+    class CleanupTimerTask implements Runnable {
+        private final Logger log = getLogger(getClass());
+        private String sessionId;
+        private AaaManager aaaManager;
+        CleanupTimerTask(String sessionId, AaaManager aaaManager) {
+            this.sessionId = sessionId;
+            this.aaaManager = aaaManager;
+        }
+
+        @Override
+        public void run() {
+            StateMachine stateMachine = StateMachine.lookupStateMachineBySessionId(sessionId);
+            if (null != stateMachine) {
+                boolean noTrafficWithinThreshold = System.currentTimeMillis() -
+                        stateMachine.getlastEapolRadiusPacketReceivedTime() >
+                (cleanupTimerTimeOutInMins * 60 * 60) / 2;
+                if ((TIMEOUT_ELIGIBLE_STATES.contains(stateMachine.state())) && noTrafficWithinThreshold) {
+                    log.info("Deleting StateMachineMapping for sessionId: {}", sessionId);
+                    cleanupTimer = null;
+                    if (stateMachine.state() == STATE_PENDING) {
+                        aaaManager.aaaStatisticsManager.getAaaStats().increaseTimedOutPackets();
+
+                    }
+                    deleteStateMachineId(sessionId);
+                    deleteStateMachineMapping(stateMachine);
+
+                }
+                else {
+                    aaaManager.scheduleTimer(sessionId, stateMachine);
+
+                }
+            } else {
+                log.info("state-machine not found for sessionId: {}", sessionId);
```

Comment:
Put it as warning/error. Ideally, this should not happen. If the session has been completed, the timer would have been cancelled and hence this log should not appear at all.

Reply:
Done